src/hal/src/pso/descriptor.rs

```diff
@@ -45,15 +45,18 @@ pub enum DescriptorType {
     UniformBuffer = 6,
     /// Read-Write, structured buffer.
     StorageBuffer = 7,
-    ///
+    /// A uniform buffer that can be indexed into when binding its associated descriptor
```

Comment:
I wouldn't say "indexed into". It's a buffer that the user can has to provide an offset for on each binding, allowing them to use different parts of the buffer for different draw calls with the same descriptor set